File: Array/Can Place Flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """
        count = 0
        flowerbed = [0] + flowerbed + [0]
        for i in range(1, len(flowerbed)-1):
            if flowerbed[i] == 0 and flowerbed[i-1] + flowerbed[i+1] == 0:
                count += 1
                flowerbed[i] = 1
        return count >= n

File: Array/Can Place Flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int count=0;
        vector<int>::iterator begin = flowerbed.begin();
        flowerbed.insert(begin, 0);
        flowerbed.push_back(0);
        for(int i=1; i<flowerbed.size()-1; i++){
            if(flowerbed[i]==0 && flowerbed[i-1] + flowerbed[i+1]==0){
                count += 1;
                flowerbed[i] = 1;
            }
        }
        return count >= n;
    }
};